File: Schedule/Schedule/Inputs/GetTeacherInput.cs
using Schedule.Entities;

namespace Schedule.Inputs;

public class GetTeacherInput
{
    public string? Search { get; set; }
    public DayOfWeek? WeekDay { get; set; }
    public LessonNumber? LessonNumber { get; set; }
    public int? LessonInfoId { get; set; }
}

File: client/src/app/core/inputs/get-teacher-input.ts
import {DayOfWeek} from "../enums/dayOfWeek";
import {LessonNumberString} from "../enums/lessonNumber";

export interface GetTeacherInput {
  search: string | null;
  weekDay: DayOfWeek | null;
  lessonNumber: LessonNumberString | null;
  lessonInfoId: number | null;
}
